app/models/user.rb

```diff
@@ -20,6 +21,17 @@ def initialize(attributes = {})
     person.email = email
   end
 
+  def roles=(roles)
+    roles = [*roles].map { |r| r.to_sym }
+    self.roles_mask = (roles & ROLES).map { |r| 2**ROLES.index(r) }.inject(0, :+)
```

Comment:
You might want to add a comment on what this line/method accomplishes

Reply:
not needed anymore, deleted